### PR TITLE
JSUI-2675 Add CategoryFacetValue

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Will add 'values' (management of multiple values) in a future PR, let's not make this too long
https://coveord.atlassian.net/browse/JSUI-2675

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/CategoryFacet/CategoryFacetTestUtils.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { ICategoryFacetValue } from '../../../src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValue';
```
Comment (olamothe) [inline unitTests/ui/CategoryFacet/CategoryFacetTestUtils.ts]: Any reason why you did not put all those test utility function in the `Fake` module ?

File: unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      initializeComponent();
+    });
+
+    function initializeComponent(options?: ICategoryFacetOptions) {
+      facet = CategoryFacetTestUtils.createFakeFacet(options);
+      facetValue = new CategoryFacetValue(CategoryFacetTestUtils.createFakeFacetValues(1)[0], facet);
+      facetValueRenderer = new CategoryFacetValueRenderer(facetValue, facet);
+    }
+
+    it('should render without errors', () => {
```
Comment (olamothe) [inline unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts]: Should add XSS testing, for all possible fields contained in the API response on a category facet.

The index facet value, or display value, returned by the API could contain harmful content (html or JS code), and should not be trusted by the UI.

File: src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValue.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import * as Globalize from 'globalize';
+import { CategoryFacet } from '../CategoryFacet';
+import { FacetValueState } from '../../../rest/Facet/FacetValueState';
+import { CategoryFacetValueRenderer } from './CategoryFacetValueRenderer';
+import { l } from '../../../strings/Strings';
+
+export interface ValueRenderer {
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValue.ts]: Should this interface name start with an `I`? e.g. `IValueRenderer`

File: src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValue.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export interface ICategoryFacetValue {
+  value: string;
+  path: string[];
+  displayValue: string;
+  state: FacetValueState;
+  numberOfResults: number;
+  moreValuesAvailable: boolean;
+  preventAutoSelect?: boolean;
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValue.ts]: Why is this property optional?

File: src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRenderer.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { CategoryFacet } from '../CategoryFacet';
+import { CategoryFacetValue, ValueRenderer } from './CategoryFacetValue';
+
+export class CategoryFacetValueRenderer implements ValueRenderer {
+  private dom: Dom;
+
+  constructor(private facetValue: CategoryFacetValue, private facet: CategoryFacet) { }
+
+  public render() {
+    this.dom = $$('li', {
```
Comment (samisayegh) [inline src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRenderer.ts]: Should we add certain accessibility attributes to each li? If so, this could be included here on in a separate PR focused on accessibility.

File: unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  describe('CategoryFacetValueRendererTest', () => {
+    let facetValue: CategoryFacetValue;
+    let facetValueRenderer: CategoryFacetValueRenderer;
+    let facet: CategoryFacet;
+
+    beforeEach(() => {
+      initializeComponent();
+    });
+
+    function initializeComponent(options?: ICategoryFacetOptions) {
```
Comment (samisayegh) [inline unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts]: I find a more elegant pattern is to place the `options` object in the outer-most `describe` block scope (e.g. line 11). The object is reset in the `beforeEach`, and each tests changes properties as needed. When calling `initializeComponent`, there is no need to pass the object, since the function can retrieve the `describe` block instance.

All that being said, at the moment, it looks like the `options` arg is not being used, so this is more for a larger test suite where the options need to be individually changed.

File: src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRenderer.ts (hunk 5 above)
Comment (ThibodeauJF) [reply src/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRenderer.ts]: Yes it's basic displaying for now and I will add all the necessary style + accessibility in another PR

File: unitTests/ui/CategoryFacet/CategoryFacetTestUtils.ts (hunk 1 above)
Comment (ThibodeauJF) [reply unitTests/ui/CategoryFacet/CategoryFacetTestUtils.ts]: Seems more appropriate in something only for facets instead of a generic fake module, same for DynamicFacet...

File: unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts (hunk 2 above)
Comment (ThibodeauJF) [reply unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts]: I'll put a todo and will elaborate renderer in a future PR

File: unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts (hunk 6 above)
Comment (ThibodeauJF) [reply unitTests/ui/CategoryFacet/CategoryFacetValues/CategoryFacetValueRendererTest.ts]: There are default options in the Utils, this is to pass an object to override/add new options, but I'll removve it for now
